tests/test_bench_config.py

```diff
@@ -201,62 +221,3 @@ def test_read_config():
     res = read_config(filename)
     os.remove(filename)
     assert config == res
```

Comment:
You removed a lot of tests around checking if the default values are set correctly when keys aren't provided. Why?

Reply:
Yes I made it so we no longer set default values in the config dictionary. I did this because I wanted to define most of the default values in the `__init__` method of the `CableRepository` class so that it was clear what the default values were just by looking at the class.

For the `science_configurations` key, the default value is imposed in benchcab.py by using the `get()` method:
```python
self.config.get("science_configurations", internal.DEFAULT_SCIENCE_CONFIGURATIONS)
```

But I'm glad you pointed this out because it made me think of two things:

1. I have changed the default values for the realisation specific optional parameters to be `None` (which maps to `null` in YAML), i.e:
```python
name: Optional[str] = None,
revision: Optional[int] = None,
patch: Optional[dict] = None,
build_script: Optional[str] = None,
```
This would be a breaking API change and I have forgotten to update the documentation. Should we keep the default values the same as before with `revision=-1, patch={}, build_script=""`? I personally think using `None` is better, especially for `revision` since `-1` is a truthy value.

2. I should probably be testing the `__init__` method in a separate test and asserting the correct default values are set.

Reply:
Right, sorry missed that re-organisation amidst the other changes. Using `None` for default is fine with me.
And definitely do not write a test for the `__init__` method

Reply:
Ok, I will update the documentation.

Reply:
Documentation has been updated here: [8acd685](https://github.com/CABLE-LSM/benchcab/pull/99/commits/8acd685fb96ba57d43b672e562847a8c1661c9c7)